### PR TITLE
fix: switching from ct to e2e (non-configured) does not go through setup

### PR DESCRIPTION
**Pull request by ZachJW34**
- Closes [UNIFY-1791](https://cypress-io.atlassian.net/browse/UNIFY-1791)

### User facing changelog
Navigate to e2e project setup when switching from CT (configured) to e2e (not-configured) from App

### Additional details
We were missing a call to `scaffoldTestingType()` when switching testing types from the App. This function is what initializes e2e setup and causes the Launchpad to render the e2e setup flow. I pulled the logic out into it's own function so I could call this from within the added Launchpad `project-setup.cy.ts` tests to verify we are redirecting as expected.

### How has the user experience changed?
Before:

https://user-images.githubusercontent.com/25158820/169905004-437adabd-d8ae-4b15-9d46-8e5957b88131.mov

After:

https://user-images.githubusercontent.com/25158820/169905032-c30b6498-f40e-4b9b-8d50-90772891bc8c.mov

### PR Tasks

- [x] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [ ] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

File: packages/graphql/src/schemaTypes/objectTypes/gql-Mutation.ts [hunk 1]
```diff
@@ -606,33 +606,12 @@ export const mutation = mutationType({
...
-        ctx.project.setRelaunchBrowser(ctx.lifecycleManager.isTestingTypeConfigured(args.testingType))
-        ctx.actions.project.setAndLoadCurrentTestingType(args.testingType)
-        await ctx.actions.project.reconfigureProject()
+        await ctx.actions.project.switchTestingTypesAndRelaunch(args.testingType)
 
         return true
       },
     })
 
-    t.field('setTestingTypeAndReconfigureProject', {
```
Comment (tbiethman) [inline packages/graphql/src/schemaTypes/objectTypes/gql-Mutation.ts:617]: Was this dead code?

File: packages/launchpad/cypress/e2e/project-setup.cy.ts [hunk 2]
```diff
@@ -629,4 +629,46 @@ describe('Launchpad: Setup Project', () => {
...
+      cy.get('[data-cy-testingtype="e2e"]').click()
+      cy.contains('h1', 'Choose a Browser')
+
+      // Execute same function that is called in the browser to switch testing types
+      cy.withCtx(async (ctx, { sinon }) => {
+        sinon.stub(ctx.actions.browser, 'closeBrowser')
+        sinon.stub(ctx.actions.electron, 'refreshBrowserWindow')
+        sinon.stub(ctx.actions.electron, 'showBrowserWindow')
+        await ctx.actions.project.switchTestingTypesAndRelaunch('component')
+        ctx.emitter.toLaunchpad()
```
[the remark this answers is not in the document]
Comment (ZachJW34) [reply packages/launchpad/cypress/e2e/project-setup.cy.ts:668]: This was required though I couldn't track down exactly why. To be more in line with the actual behavior, I'm going to use `cy.reload` as that is what the `ctx.actions.electron.refreshBrowserWindow()` would trigger outside of a test.

**Comment by ZachJW34** [comment]
@tbiethman latest commit has no failing launchpad tests

**Review by tbiethman — APPROVED** [review]

**Review by marktnoonan — APPROVED** [review]
Super!

**Review by tgriesser — APPROVED** [review]

**Pull request merged**